sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/FormRecognizerAsyncClient.java

```diff
@@ -203,21 +422,160 @@ public FormRecognizerServiceVersion getServiceVersion() {
         };
     }
 
-    private Function<PollingContext<OperationResult>, Mono<IterableStream<ExtractedReceipt>>>
+    private Function<PollingContext<OperationResult>, Mono<IterableStream<RecognizedReceipt>>>
         fetchExtractReceiptResult(boolean includeTextDetails) {
         return (pollingContext) -> {
             final UUID resultUid = UUID.fromString(pollingContext.getLatestResponse().getValue().getResultId());
             return service.getAnalyzeReceiptResultWithResponseAsync(resultUid)
-                .map(modelSimpleResponse -> toReceipt(modelSimpleResponse.getValue().getAnalyzeResult(),
-                    includeTextDetails));
+                .map(modelSimpleResponse -> {
+                    throwIfAnalyzeStatusInvalid(modelSimpleResponse);
+                    return toReceipt(modelSimpleResponse.getValue().getAnalyzeResult(),
+                        includeTextDetails);
+                });
+        };
+    }
+
+    private Function<PollingContext<OperationResult>, Mono<OperationResult>> contentAnalyzeActivationOperation(
```

Comment:
Other than the activation method and the result are these poller helper methods different? Could it be combined into a single helper method?

Reply:
Would like to consider doing this post preview 1 - https://github.com/Azure/azure-sdk-for-java/issues/10187